src/Chunk.ts

```diff
@@ -825,6 +843,126 @@ export default class Chunk {
 		};
 	}
 
+	getRenderedSourceLength() {
+		return this.renderedSource.toString().length;
+	}
+
+	/*
+	 * Performs a full merge of another chunk into this chunk
+	 * chunkList allows updating references in other chunks for the merged chunk to this chunk
+	 * A new facade will be added to chunkList if tainting exports of either as an entry point
+	 */
+	merge(chunk: Chunk, chunkList: Chunk[], options: OutputOptions) {
```

Comment:
Initially that was the way I was handling it, but because this needs to access all the internal private properties, and because it is mutating of the chunk (we don't spit out a third chunk, we merge into the first input), I think it is better implemented on Chunk.